### PR TITLE
1718/bnd maven plugin

### PR DESCRIPTION
**Pull request by davidjgonzalez**
Update project to use bnd-maven-plugin

### CODE REVIEW

**Comment by davidjgonzalez** [comment]
@adamcin when verifying my build, oakpal is complaining: 

```
[ERROR] Failed to execute goal net.adamcin.oakpal:oakpal-maven-plugin:2.1.0:scan (oakpal-scan) on project acs-aem-commons-ui.apps: Failed to execute package scan. Failed to resolve cndNames.: Failed to find node type definition on classpath for cndName nodetypes/nodetypes.cnd -> [Help 1]
```

however the node this path exists (no changes to it from master) .. `<project root>/src/test/resources/nodetypes/nodetypes.cnd `

Any idea why? I'm guessing its something simple I'm overlooking... 🤦

**Comment by davidjgonzalez** [comment]
@kwin / @joerghoh - if you get a moment cant you take a quick pre-review look at this? Your maven-foo is stronger than mine and want to make sure I didn't botch syntax (esp in CDATA) and/or there isn't a more efficient configuration.

File: bundle/pom.xml [hunk 1]
```diff
@@ -7,100 +7,111 @@
...
+                !android.util.*, \
+                <!-- shaded libraries --> \
+                !com.google.common.*, \
+                !com.google.thirdparty.*, \
+                !com.jcraft.*, \
+                !io.jsonwebtoken.*, \
+                *
+-exportcontents: ${packages;VERSIONED}
+-snapshot: ${tstamp;yyyyMMddHHmmssSSS}
+Bundle-DocURL:
```
Comment (kwin) [inline bundle/pom.xml:76]: I would move the -exportcontents, -snapshot and Bundle-DocURL instructions/headers to the parent pom's bnd configuration (it is not specific to this bundle)

File: bundle/pom.xml [hunk 2]
```diff
@@ -7,100 +7,111 @@
...
+                !org.checkerframework.checker.nullness.qual, \
+                !android.util.*, \
+                <!-- shaded libraries --> \
+                !com.google.common.*, \
+                !com.google.thirdparty.*, \
+                !com.jcraft.*, \
+                !io.jsonwebtoken.*, \
+                *
+-exportcontents: ${packages;VERSIONED}
+-snapshot: ${tstamp;yyyyMMddHHmmssSSS}
```
Comment (kwin) [inline bundle/pom.xml:75]: Probably we can stay with SNAPSHOT as recommended in https://github.com/bndtools/bnd/tree/master/maven/bnd-maven-plugin#reproducible-builds.

File: bundle/pom.xml [hunk 3]
```diff
@@ -7,100 +7,111 @@
...
+Bundle-DocURL:
+-plugin org.apache.sling.caconfig.bndplugin.ConfigurationClassScannerPlugin
+-plugin org.apache.sling.bnd.models.ModelsScannerPlugin
```
Comment (kwin) [inline bundle/pom.xml:78]: we still need to Felix SCR plugin as long as we are not yet on the OSGi annotations

File: bundle/pom.xml (hunk 1 above)
Comment (kwin) [inline bundle/pom.xml:76]: The Bundle-DocUrl should point to the Github site.

File: pom.xml [hunk 4]
```diff
@@ -141,6 +142,17 @@
...
+                    <groupId>biz.aQute.bnd</groupId>
+                    <artifactId>bnd-maven-plugin</artifactId>
+                    <version>${bnd.version}</version>
+                </plugin>
+                <plugin>
+                    <groupId>biz.aQute.bnd</groupId>
+                    <artifactId>bnd-baseline-maven-plugin</artifactId>
+                    <version>${bnd.version}</version>
+                </plugin>
                 <!-- all custom plugins -->
```
Comment (kwin) [inline pom.xml:156]: maven-scr-plugin needs to be removed (it is not compatible with bnd-maven-plugin)

Comment (kwin) [reply pom.xml:156]: remove  maven-bundle-plugin!

**Review by kwin — CHANGES_REQUESTED** [review]

**Comment by adamcin** [comment]
> @adamcin when verifying my build, oakpal is complaining:
> 
> ```
> [ERROR] Failed to execute goal net.adamcin.oakpal:oakpal-maven-plugin:2.1.0:scan (oakpal-scan) on project acs-aem-commons-ui.apps: Failed to execute package scan. Failed to resolve cndNames.: Failed to find node type definition on classpath for cndName nodetypes/nodetypes.cnd -> [Help 1]
> ```
> 
> however the node this path exists (no changes to it from master) .. `<project root>/src/test/resources/nodetypes/nodetypes.cnd `
> 
> Any idea why? I'm guessing its something simple I'm overlooking... 🤦

@davidjgonzalez Can you confirm that the cnd is being copied to `target/test-classes/nodetypes/nodetypes.cnd`?

**Comment by davidjgonzalez** [comment]
@adamcin I do not see these `target/test-classes/nodetypes/nodetypes.cnd` in my `ui.apps` project.

I actually tried doing an `mvn clean install` from master and ran into the same issue.

**Comment by davidjgonzalez** [comment]
@adamcin so this is strange (and maybe why i haven't seen this before). Everything seems to work in intellij, however when i run the command from the command line it fails.
My version of maven is the same 3.6.3 (command line is 3.6.3_1), and I'm using the same version of java. That help ring any bells?

**Comment by davidjgonzalez** [comment]
Closing for now. To pick up post 5.0

**Pull request closed without merging**